Add a C API test for appending

noahgoldman opened the pull request:
This adds a test for appending to a file by writing, closing the file, opening it again and seeking to the end, then writing again.  I added a clarifying comment before the test describing why this method is used as opposed to just opening the file with "O_APPEND".

@kstinsonqc @mckurt @mikeov

mckurt commented on src/cc/qfsc/test-qfsc.c:
```diff
@@ -339,6 +339,50 @@ static char* test_qfs_get_data_locations() {
+//
+// Using O_APPEND itself does not produce the same result.  Opening a file with
+// O_APPEND and writing to it will create hole of the appropriate size to make
+// the previous chunk reach the chunksize.  This can be useful as an atomic
```
"This can be useful as an atomic append, but is not the behavior you'd expect from a normal filesystem."

I don't think this statement is needed here.

mckurt commented:
@noahgoldman Can you squeeze this into a single commit? One that reads something like "Add 	an append test by writing, closing, then seeking and writing ...".

mckurt commented on src/cc/qfsc/test-qfsc.c:
```diff
@@ -339,6 +339,50 @@ static char* test_qfs_get_data_locations() {
+  strcpy(expected_str, testdata);
+  strcat(expected_str, testdata);
+
+  char buf[expected_len];
```
not sure variable sized array declaration is a good idea.

noahgoldman commented:
@mckurt Updated this PR from your comments:

* Rebased into a single commit and changed the message
* Removed the ""This can be useful as an atomic append..." comment
* Replaced `strcpy` and `strcat` with two calls of `memcpy`
* Changed the `qfs_read` buffer to initialize with size 4096
